pkgs/tools/audio/beets/common.nix

```
@@ -55,7 +56,17 @@ let
   };
 
   basePlugins = lib.mapAttrs (_: a: { builtin = true; } // a) (import ./builtin-plugins.nix inputs);
-  allPlugins = lib.mapAttrs (n: a: mkPlugin { name = n; } // a) (lib.recursiveUpdate basePlugins pluginOverrides);
+  isInList = x: l: (lib.count (lItem: x==lItem) l) > 0;
+  pluginOverrides' = lib.mapAttrs
+    (plugName: plugVal: lib.throwIf
+      (isInList plugName deprecatedPlugins)
+      "beets evaluation error: Plugin ${plugName} was enabled in pluginOverrides, but it was deprecated. Remove it to fix evaluation."
+      plugVal
+    )
+    pluginOverrides
+  ;
```

I find `isInList` a bit confusing. What's the reason you're not using `lib.elem` (which is an alias to `builtins.elem`)? Also, you should be able omit the `plugVal` and just curry the inner function.

```suggestion
  pluginOverrides' = lib.mapAttrs
    (plugName: lib.throwIf
      (lib.elem plugName deprecatedPlugins)
      "beets evaluation error: Plugin ${plugName} was enabled in pluginOverrides, but it was deprecated. Remove it to fix evaluation."
    )
    pluginOverrides
  ;
```

---

Thanks for the code suggestion! [I didn't find `lib.elem` from some reason in noogle.dev](https://github.com/nix-community/noogle/issues/227) so I invented it myself. I'm a bit confused how does your code works, but in anycase it is better.